Userland/Libraries/LibJS/Runtime/Object.h

```diff
@@ -213,11 +213,11 @@ class Object : public Cell {
     enum class ConstructWithoutPrototypeTag { Tag };
     enum class ConstructWithPrototypeTag { Tag };
 
-    Object(GlobalObjectTag, Realm&);
-    Object(ConstructWithoutPrototypeTag, Realm&);
-    Object(Realm&, Object* prototype);
-    Object(ConstructWithPrototypeTag, Object& prototype);
-    explicit Object(Shape&);
+    Object(GlobalObjectTag, Realm&, bool may_interfere_with_indexed_property_access = false);
+    Object(ConstructWithoutPrototypeTag, Realm&, bool may_interfere_with_indexed_property_access = false);
+    Object(Realm&, Object* prototype, bool may_interfere_with_indexed_property_access = false);
+    Object(ConstructWithPrototypeTag, Object& prototype, bool may_interfere_with_indexed_property_access = false);
+    explicit Object(Shape&, bool may_interfere_with_indexed_property_access = false);
```

Comment:
Please use an enum instead of a bool for this, to keep call sites readable.
Something like:
```cpp
enum class MayInterfereWithIndexedPropertyAccess {
    No,
    Yes,
};
```
The member can still be a bool ofc. :)